providers/implementations/signature/rsa.c

```diff
@@ -1340,14 +1339,22 @@ static const OSSL_PARAM known_settable_ctx_params[] = {
     OSSL_PARAM_END
 };
 
+static const OSSL_PARAM settable_ctx_params_no_digest[] = {
+    OSSL_PARAM_utf8_string(OSSL_SIGNATURE_PARAM_PAD_MODE, NULL, 0),
+    OSSL_PARAM_utf8_string(OSSL_SIGNATURE_PARAM_MGF1_DIGEST, NULL, 0),
+    OSSL_PARAM_utf8_string(OSSL_SIGNATURE_PARAM_MGF1_PROPERTIES, NULL, 0),
+    OSSL_PARAM_utf8_string(OSSL_SIGNATURE_PARAM_PSS_SALTLEN, NULL, 0),
+    OSSL_PARAM_END
+};
+
 static const OSSL_PARAM *rsa_settable_ctx_params(void *vprsactx,
                                                  ossl_unused void *provctx)
 {
     PROV_RSA_CTX *prsactx = (PROV_RSA_CTX *)vprsactx;
 
     if (prsactx != NULL && !prsactx->flag_allow_md)
-        return &known_settable_ctx_params[2];
-    return known_settable_ctx_params;
+        return settable_ctx_params_no_digest;
```

Comment:
I don't know detailed we need to be here but the RSA key type would be a different subset.

Reply:
Actually not. This is not about the key type but about the PAD_MODE - if that is PSS, then you can set those additional properties. But I do not think we want to modify the list based on which PAD_MODE is set a the point of time in the context.